### ISSUE TITLE
Redis#connect should return promise

### ISSUE THREAD

**Issue by Shumkov** [luin/ioredis#30]
I want to handle redis connection error. It will be very useful:
```javascript
var redisClient = new Redis({ lazyConnect: true });
redisClient.connect().then(function() {
    console.log('Yehhhuuu!');
}, function(error) {
    console.error(error);
});
```

**Comment by joshuairl** [comment]
I agree.
At least some sort of event that fires upon connection because intermittently I'm getting strange issues when server peaks and it takes it a bit longer to respond to a connection.

In basic tests, it is connected and ready instantly but in some cases even a few ms difference could mean whether or not it's instantiated or not.

**Comment by AVVS** [comment]
https://github.com/luin/ioredis#connection-events

**Comment by joshuairl** [comment]
Sorry about this, I did realize this was already done after I posted this.
Should have updated my comment.

Thanks again for all you do!

**Comment by luin** [comment]
Released in 1.2.7
